codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -384,4 +373,71 @@ protected ZlibWrapper determineWrapper(String acceptEncoding) {
         }
         return null;
     }
+
+    private CompressionEncoderFactory getEncoderFactory(String contentEncoding) {
+        for (CompressionEncoderFactory encoderFactory: factories) {
```

Comment:
consider storing  the `CompressionEncoderFactory` in a `Map` and so make the lookup O(1)